spec/controllers/api/v1/accounts/agents_controller_spec.rb

```diff
@@ -173,5 +174,63 @@
         expect(account.users.last.name).to eq('NewUser')
       end
     end
+
+    context 'when the account has reached its agent limit' do
+      params = { name: 'NewUser', email: Faker::Internet.email, role: :agent }
+
+      before do
+        account.update(limits: { agents: 4 })
+        create_list(:user, 4, account: account, role: :agent)
+      end
+
+      it 'prevents adding a new agent and returns a payment required status' do
+        post "/api/v1/accounts/#{account.id}/agents", params: params, headers: admin.create_new_auth_token, as: :json
+
+        expect(response).to have_http_status(:payment_required)
+        expect(response.body).to include('Account limit exceeded. Please purchase more licenses')
+      end
+    end
+  end
+
+  describe 'POST /api/v1/accounts/{account.id}/agents/bulk_create' do
+    let(:emails) { ['test1@example.com', 'test2@example.com', 'test3@example.com'] }
+    let(:bulk_create_params) { { emails: emails } }
+
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        post "/api/v1/accounts/#{account.id}/agents/bulk_create", params: bulk_create_params
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when authenticated as admin' do
+      context 'when there is sufficient limit' do
+        it 'creates multiple agents successfully' do
+          # Assume the limit is more than 3 for this test
+          account.update(limits: { agents: 10 })
+
+          expect do
+            post "/api/v1/accounts/#{account.id}/agents/bulk_create", params: bulk_create_params, headers: admin.create_new_auth_token
+          end.to change(User, :count).by(3)
+
+          expect(response).to have_http_status(:ok)
+        end
+      end
+
+      context 'when exceeding agent limit' do
+        it 'prevents creating agents and returns a payment required status' do
+          # Set the limit to be less than the number of emails
+          account.update(limits: { agents: 2 })
+
+          expect do
+            post "/api/v1/accounts/#{account.id}/agents/bulk_create", params: bulk_create_params, headers: admin.create_new_auth_token
+          end.not_to change(User, :count)
+
+          expect(response).to have_http_status(:payment_required)
+          expect(response.body).to include('Account limit exceeded. Please purchase more licenses')
+        end
+      end
```

Comment:
should move the limits related to spec to enterprise spec. since in community edition the limits are static 
ref: https://github.com/chatwoot/chatwoot/blob/39e27d2a23b2dd2e455d3c04ac6c2dc6fb9a6d73/app/models/account.rb#L121